src/yetibot/api/aws.clj

```diff
@@ -0,0 +1,175 @@
+(ns yetibot.api.aws
+  (:require
+    [yetibot.core.schema :refer [non-empty-str]]
+    [schema.core :as sch]
+    [yetibot.core.config :refer [get-config]]
+    [cognitect.aws.client.api :as aws]))
+
+(def aws-schema
+  {:aws-access-key-id         non-empty-str
+   :aws-secret-access-key     non-empty-str
+   (sch/optional-key :region) non-empty-str
+   (sch/optional-key :dev-mode) sch/Bool})
+
+(defn config
+  "Returns AWS-related configuration"
+  []
+  (:value (get-config aws-schema [:aws])))
+
+(defn configured? [] (config))
+
+; aws API credentials
+(def aws-access-key-id (:aws-access-key-id (config)))
+(def aws-secret-access-key (:aws-secret-access-key (config)))
+(def region (:region (config)))
+(def dev-mode? (:dev-mode (config)))
+
+(defn make-aws-client
+  "Returns an aws client given an aws keywordized service name (:iam, :ec2, :s3 etc.)"
+  [service-name]
+  (let [client (aws/client {:api                  service-name
+                            :credentials-provider (cognitect.aws.credentials/basic-credentials-provider
+                                                    {:access-key-id     aws-access-key-id
+                                                     :secret-access-key aws-secret-access-key})
+                            :region               region})]
+    (when (and (not (nil? dev-mode?))
+               (true? dev-mode?))
```

Comment:
This feels little cumbersome, and will require repetition if need to reference `dev-mode?` again.  Could you handle this earlier, when pulling out of `config`, on L25?

Reply:
no problem, I will fix this
thanks for the feedback ;-)